### PR TITLE
Refactor stamp.py

### PR DESCRIPTION
**Pull request by esheldon**
Pull out some useful functionality.
    
    Useful stuff for calculating stamp sizes was put into stamp_utils.py
    
        get_stamp_size (was in LSST_SiliconBuilder.__init__)
            get_star_stamp_size
            get_gal_stamp_size
        get_good_phot_stamp_size (was LSST_SiliconBuilder._getGoodPhotImageSize)
        get_good_phot_stamp_size1 (was LSST_SiliconBuilder._getGoodPhotImageSize1)
    
    Useful stuff for psfs was put into psf_utils.py
    
        make_double_gaussian (was LSST_SiliconBuilder.DoubleGaussian)
        make_kolmogorov_and_gaussian_psf was (LSST_SiliconBuilder.Kolmogorov_and_Gaussian_PSF)
        make_fft_psf (was LSST_SiliconBuilder.make_fft_psf)

Add a .flake8 to avoid some style complaints

### CODE REVIEW

**Comment by esheldon** [comment]
Who wants to take this one?

**Comment by esheldon** [comment]
The data download failed in the CI.  I don't have permission to rerun the workflow, can someone please kick that off?

File: imsim/stamp_utils.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+    )
+    stamp_size = psf.getGoodImageSize(pixel_scale)
+    # No point in this being larger than a CCD.  Cut back to Nmax if larger than this.
+    stamp_size = min(stamp_size, Nmax)
+    return stamp_size
+
+
+def get_gal_stamp_size(obj_achrom, nominal_flux, noise_var, Nmax, pixel_scale):
+    """
+    Get a stamp size for a star (DeltaFunction) object
```
Comment (rmjarvis) [inline imsim/stamp_utils.py:160]: This doc is wrong.

**Review by rmjarvis — APPROVED** [review]
LGTM.  Just one doc typo.

**Comment by esheldon** [comment]
Is this OK to merge now?

**Comment by rmjarvis** [comment]
It's fine by me, but I usually let Jim or Chris make the call.

**Comment by jchiang87** [comment]
It's ok with me.

**Comment by cwwalter** [comment]
OK.

**Pull request merged**